src/Microsoft.VisualBasic/src/Microsoft/VisualBasic/FileIO/FileSystem.Windows.vb

```diff
@@ -0,0 +1,354 @@
+        ''' <param name="errorCode">An error code from winerror.h.</param>
+        ''' <remarks>
+        ''' - This method is based on sources\ndp\clr\src\BCL\System\IO\_Error.cs::WinIOError, except the following.
+        ''' - Exception message does not contain the path since at this point it is normalized.
+        ''' - Instead of using PInvoke of GetMessage and MakeHRFromErrorCode, use managed code.
+        ''' </remarks>
+        <SecurityCritical()>
+        Private Shared Sub ThrowWinIOError(ByVal errorCode As Integer)
+            Select Case errorCode
+                Case NativeTypes.ERROR_FILE_NOT_FOUND
+                    Throw New IO.FileNotFoundException()
+                Case NativeTypes.ERROR_PATH_NOT_FOUND
+                    Throw New IO.DirectoryNotFoundException()
+                Case NativeTypes.ERROR_ACCESS_DENIED
+                    Throw New UnauthorizedAccessException()
+                Case NativeTypes.ERROR_FILENAME_EXCED_RANGE
+                    Throw New IO.PathTooLongException()
+                Case NativeTypes.ERROR_INVALID_DRIVE
+                    Throw New IO.DriveNotFoundException()
+                Case NativeTypes.ERROR_OPERATION_ABORTED, NativeTypes.ERROR_CANCELLED
+                    Throw New OperationCanceledException()
+                Case Else
+                    ' Including these from _Error.cs::WinIOError.
+                    'Case NativeTypes.ERROR_ALREADY_EXISTS
+                    'Case NativeTypes.ERROR_INVALID_PARAMETER
+                    'Case NativeTypes.ERROR_SHARING_VIOLATION
+                    'Case NativeTypes.ERROR_FILE_EXISTS
+                    Throw New IO.IOException((New Win32Exception(errorCode)).Message,
+                        System.Runtime.InteropServices.Marshal.GetHRForLastWin32Error())
+            End Select
+        End Sub
+
+        Private Shared Sub WinNTCopyOrMove(sourceFileFullPath As String, destinationFileFullPath As String)
+            Try
+                Dim succeed As Boolean = NativeMethods.MoveFileEx(
+                        sourceFileFullPath, destinationFileFullPath, m_MOVEFILEEX_FLAGS)
+                ' GetLastWin32Error has to be close to PInvoke call. FxCop rule.
+                If Not succeed Then
+                    ThrowWinIOError(System.Runtime.InteropServices.Marshal.GetLastWin32Error())
+                End If
+            Catch
+                Throw
+            End Try
+        End Sub
+
+        ' When calling MoveFileEx, set the following flags:
+        ' - Simulate CopyFile and DeleteFile if copied to a different volume.
+        ' - Replace contents of existing target with the contents of source file.
+        ' - Do not return until the file has actually been moved on the disk.
+        Private Const m_MOVEFILEEX_FLAGS As Integer = CInt(
+            MoveFileExFlags.MOVEFILE_COPY_ALLOWED Or
+            MoveFileExFlags.MOVEFILE_REPLACE_EXISTING Or
+            MoveFileExFlags.MOVEFILE_WRITE_THROUGH)
```

Comment:
Please move `m_MOVEFILEEX_FLAGS` below `m_SHELL_OPERATION_FLAGS_HIDE_UI`, to match the reference source, so it's easier to verify the differences. Thanks.